.github/workflows/go-mod-auto-bump.yml

```diff
@@ -33,14 +33,29 @@ jobs:
         run: |
           echo "CURRENT_BRANCH=$(echo ${GITHUB_REF} | sed 's|refs/heads/||')" >> $GITHUB_OUTPUT
 
-      - name: Run script 
+      # This step fixes a bug, due to which the push to any branch, for example: to person-A/name, triggered this workflow on branch person-b/name,
+      # and updated the go.mod on branch person-B/name with commit from branch person-A/name
+      - name: Check if the current branch contains commit that triggered the workflow
+        id: valid_branch 
+        run: |
+          CURRENT_BRANCH=${{ steps.extract_branch.outputs.CURRENT_BRANCH }}
+          COMMIT_EXISTS_ON=$(echo $(git branch --contains ${{ github.event.after}} ) | grep $CURRENT_BRANCH)
+
+          if [[ "$COMMIT_EXISTS_ON" == *"$CURRENT_BRANCH"* ]];
+          then
+            echo "VALID=1" >> $GITHUB_OUTPUT
+          fi
+          
+      - name: Run script
+        if: ${{ steps.valid_branch.outputs.VALID }} == 1
```

Comment:
So assume I'm on branch `roman/A` and you are on `ruslan/B`

Then, I modify `osmoutils` on my branch. Then, this workflow will run for both `roman/A` and `ruslan/B`. However, it will not continue past this step on `ruslan/B`. Is my understanding correct?

I'm wondering if there is a way to not run this job on other branches at all to avoid having a queue in CI.

Let me know what you think

Reply:
Yes, that is right. However, as mentioned [here](https://github.com/osmosis-labs/osmosis/issues/4660#issuecomment-1475526794), I haven't seen such behavior by changing, for example, `osmoutils` on my local branch. I checked the `Action` tab, which showed me, that this workflow was running only for my branch. 

A more detailed explanation you may find in the attached link above

Reply:
let me know what you think on removing `main` from this workflow triggers